go/workflow-interceptor-updates.md

```diff
@@ -0,0 +1,269 @@
+# Go SDK Workflow Interceptor Updates and OpenTelemetry Tracing Support
+
+This proposal is for updates to Go workflow interceptors with the following features:
+
+* Support for and implementation of OpenTelemetry tracing
```

Comment:
We should make sure that interceptors can be used for authz too (part of the reason we add an explicit `error` to the return type).

Reply:
I added a note that examples will be added during implementation showing header alteration in interceptors and client werappers...are headers enough?

Reply:
The use-case I know of is (example for StartWorkflow):
1. Client sends StartWorkflow request with auth token in header to a proxy that sits in front of the server (or modified server)
1. Server translates token to auth claim
2. Worker picks up StartWorkflow task, compares request and claims and either processes or rejects the task

Reply:
I haven't seen an example just heard about this practice.